warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java

```diff
@@ -602,16 +614,43 @@ public List<datawave.data.type.Type<?>> getDataTypes(String fieldName) {
                 compilePatterns();
             }
 
+            List<Integer> patternLengths = new ArrayList<>();
+            int bestMatch = 0;
+            Collection<datawave.data.type.Type<?>> bestMatchTypes = null;
+
             for (Matcher patternMatcher : typeCompiledPatternMap.keySet()) {
+                Collection<datawave.data.type.Type<?>> patternTypes = typeCompiledPatternMap.get(patternMatcher);
 
                 if (patternMatcher.reset(fieldName).matches()) {
-                    Collection<datawave.data.type.Type<?>> patternTypes = typeCompiledPatternMap.get(patternMatcher);
-                    types.addAll(patternTypes);
-                    typeFieldMap.putAll(fieldName, patternTypes);
+
+                    if (useMostPreciseFieldTypeRegex) {
+                        int patternLength = patternMatcher.pattern().toString().length();
+                        if (patternLengths.contains(patternLength)) {
```

Comment:
I think we only care if we have 2 or more patterns with the bestMatch length.

Reply:
OK, perhaps a warning in any case is good.  Lets update the comment to include that this implies that the normalizer for only one of them will be used.  Also this may be acceptible to the user if we have the same normalizer for each of them.  Hence log.error should be log.warn instead.